File: 1/1-3.ts
function sumOfSquaresOfTwoLargest(a: number, b: number, c: number): number {
    // Find the smallest value among a, b, and c
    const minVal = Math.min(a, b, c);
    // Compute the sum of the squares of all three numbers, then subtract the square of the smallest one
    return a * a + b * b + c * c - minVal * minVal;
}

console.log(sumOfSquaresOfTwoLargest(3, 2, 1));  // Should print 13

File: 1/1-3.rs
fn sum_of_squares_of_two_largest(a: f64, b: f64, c: f64) -> f64 {
    // Find the smallest value among a, b, and c
    let min_val = a.min(b).min(c);
    // Compute the sum of the squares of all three numbers, then subtract the square of the smallest one
    a * a + b * b + c * c - min_val * min_val
}

fn main() {
    println!("{}", sum_of_squares_of_two_largest(3.0, 2.0, 1.0));  // Should print 13.0
}
